app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -51,6 +58,92 @@ angular.module('topologyApp', ['kubernetesUI', 'ui.bootstrap'])
     $interval.cancel(promise);
   });
 
+
+  var refreshProvider = function(d) {
+    $scope.refreshProvider.value = true;
+    document.getElementsByClassName('alert alert-success alert-dismissable refresh_provider')[0].style.display ='block';
```

Comment:
> This is broken. You will have lots of fun watching angular and the browser fight.
> 
> The alert has a ng-show="refreshProvider.value", so please change that value instead of trying to do show/hide manually.
> 
> (And if it's not working, it's because d3.on('click', ... handler is not run within angular's digest cycle - just wrap any d3 callback in $timeout(callback) and you're done :).)

I *am* changing that value (refreshProvider.value) and indeed it was not working.
I also removed the data-dismiss="alert" because it was completely removing the element.
Can you explain a bit about the $timeout callback part?